core/solver/batch_dispatch.hpp

```diff
@@ -167,6 +167,41 @@ enum class log_type { simple_convergence_completion };
 }  // namespace log
 
 
+#define GKO_BATCH_INSTANTIATE_STOP(macro, ...)                          \
+    macro(__VA_ARGS__,                                                  \
+          ::gko::batch::solver::device::batch_stop::SimpleAbsResidual); \
+    template macro(                                                     \
```

Comment:
the `template` here (and in the other macros below) could be removed, if the value/index type instantiation macros would accept variable number or arguments.

Reply:
That doesn't work until C++20. A macro with `(arg, ...)` requires *two* arguments before c++20.